lib/ace/ext/papeeria/papeeria_latex_highlight_rules.coffee

```diff
@@ -0,0 +1,131 @@
+define( (require, exports, module) ->
+  'use strict'
+  oop = require('../../lib/oop')
+  TextHighlightRules = require('../../mode/text_highlight_rules').TextHighlightRules
+
+  PapeeriaLatexHighlightRules = ->
+
+    pushState = (destination) -> (currentState, stack) ->
```

Comment:
pls add a few lines of comments. What this function expects, what does it return. I mean, really the types of objects. Is `destination`are string, boolean, number? 

The chain of two functions looks unreadable. I suggest leaving only function name `pushState` and its arguments  on the first line.